Dockerfile

```
@@ -28,13 +28,34 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
 FROM base AS development
 
-COPY requirements.txt .
-
-# Install core build-time dependencies
 RUN pip install --no-cache-dir Cython setuptools wheel
 
-# Install all major dependencies from requirements.txt
-# Note: py-feat installation might still be tricky, but settle dependencies first
+# Install core stack with tight pins
+# Use numpy < 1.24 to satisfy nltools
+RUN pip install --no-cache-dir \
```

Hadolint (2.14.0)

[info] Multiple consecutive `RUN` instructions. Consider consolidation.

(DL3059)